### DISCUSSION TITLE
Build failures on Arch Linux

### DISCUSSION THREAD

**Discussion by afontenot** [TeamNewPipe/NewPipe#11612] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my question isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the question will be dismissed otherwise.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### What is/are your question(s)?

I have been trying and failing to build the project on Arch Linux (rolling release, up to date) for the last few hours. I'm attempting to test changes so I can contribute them to NewPipeExtractor.

Should be reproducible by taking the most straightforward and obvious steps. I `git clone` NewPipe and NewPipeExtractor, I'm using the most recent Android Studio from the AUR. I set my system JDK to openJDK 17, as that's what the CI builds appear to use.

The initial error I see:

> java.lang.IllegalAccessError: superclass access check failed: class org.jetbrains.kotlin.kapt3.base.javac.KaptJavaCompiler (in unnamed module @0x6cd7cf1b) cannot access class com.sun.tools.javac.main.JavaCompiler (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.main to unnamed module @0x6cd7cf1b

Searching for the issue sent me to https://stackoverflow.com/questions/77628445/java-lang-illegalaccesserror-superclass-access-check-failed-class-org-jetbrain

The suggestion here is to update Kotlin 1.9 to the latest patch version. I've done this (to 1.9.25) although I'm not sure why it's necessary as (if I read the linked bug report correctly) the issue is only supposed to happen with Java >= 21. In any event this does at least *change* the error I see.

With

    ext.kotlin_version = '1.9.25'

and syncing gradle, I get this error now:

```
> A failure occurred while executing org.jetbrains.kotlin.gradle.internal.KaptWithoutKotlincTask$KaptExecutionWorkAction
   > java.lang.reflect.InvocationTargetException (no error message)
```

Building with a stacktrace shows a root cause that seems to be internal to Kotlin:

> Caused by: java.lang.NullPointerException: Cannot read field "kind" because "<local2>" is null

[kotlin.stacktrace.txt](https://github.com/user-attachments/files/17383569/kotlin.stacktrace.txt)

**Comment by AudricV** [comment, marked as answer]
Fixed with #11636.
